### PR TITLE
Structured logging

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context
We need the pubsub adapter to use structured logging so that log entries can be searched on and used for metrics in GKE/stackdriver/grafana.

# What has changed
* Configure zap structured logger at startup
* Use zap structured logging
* Add LOG_LEVEL environment variable to set the service log level

# How to test?
Try running the service at different log levels (e.g. INFO, DEBUG, WARN, ERROR).

# Links
https://trello.com/c/OXBw9jhY/908-pubsub-adapter-structured-logging-5

## Screenshots
Example of the structured log viewed in stackdriver, shown with the right severity and the json fields showing up as expected
Screenshot 2020-05-06 at 14 22 05

### CODE REVIEW

**Comment by AdamHawtin** [comment]
[Pubsub Adapter - Structured Logging (5?)](https://trello.com/c/OXBw9jhY/908-pubsub-adapter-structured-logging-5)

File: main.go [hunk 1]
```diff
@@ -29,22 +37,22 @@ func main() {
...
-		log.Println(errors.Wrap(err, "Error indicating readiness"))
+		logger.Logger.Errorw("Error indicating readiness", "error", err)
 		shutdown(ctx, cancel, processors)
 	}
 
 	// Block until we receive OS shutdown signal or error
 	select {
 	case sig := <-signals:
-		log.Printf("OS Signal Received: %s", sig.String())
+		logger.Logger.Errorw("OS Signal Received", "signal", sig.String())
```
Comment (davidmort) [inline main.go]: this feels more like INFO not an ERROR as receiving the shutdown signal is expected

File: main.go [hunk 2]
```diff
@@ -87,7 +95,7 @@ func StartProcessors(ctx context.Context, cfg *config.Configuration, errChan cha
 
 func shutdown(ctx context.Context, cancel context.CancelFunc, processors []*processor.Processor) {
 	// cleanup for graceful shutdown
-	log.Printf("Shutting Down")
+	logger.Logger.Error("Shutting Down")
```
Comment (davidmort) [inline main.go]: similar to the comment above, this feels more like an INFO thing

File: main.go [hunk 3]
```diff
@@ -108,6 +116,6 @@ func shutdown(ctx context.Context, cancel context.CancelFunc, processors []*proc
 	//block until shutdown cancel has been called
 	<-shutdownCtx.Done()
 
-	log.Printf("CloseRabbit complete")
+	logger.Logger.Error("Shutdown complete")
```
Comment (davidmort) [inline main.go]: same as the comment above, feels like an INFO thing

**Review by davidmort — CHANGES_REQUESTED** [review]

**Review by NickGrantX — APPROVED** [review]
Looks nice 👍

**Review by davidmort — APPROVED** [review]

**Pull request merged**